pylint/extensions/redefined_loop_name.py

```diff
@@ -0,0 +1,85 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Optional checker to warn when loop variables are overwritten in the loop's body."""
+
+from astroid import nodes
+
+from pylint import checkers, interfaces
+from pylint.checkers import utils
+from pylint.checkers.variables import in_for_else_branch
+from pylint.interfaces import HIGH
+from pylint.lint import PyLinter
+from pylint.utils import utils as pylint_utils
+
+
+class RedefinedLoopName(checkers.BaseChecker):
+
+    __implements__ = interfaces.IAstroidChecker
+    name = "redefined-loop-name"
+
+    msgs = {
+        "W2901": (
+            "Redefining %r from loop (line %s)",
+            "redefined-loop-name",
+            "Used when a loop variable is overwritten in the loop body.",
+        ),
+    }
+
+    def __init__(self, linter=None):
```

Comment:
```suggestion
    def __init__(self, linter: PyLinter) -> None:
```

Weird that `mypy` doesn't complain about this. `linter` is no longer allowed to be `None`.